modules/storages/app/views/storages/project_settings/_project_folder_form.html.erb

```diff
@@ -139,20 +105,29 @@ See COPYRIGHT and LICENSE files for more details.
                          data: {
                            'project-storage-form-target': "projectFolderIdInput"
                          } %>
-      <%= styled_button_tag id: 'open_location_picker_button',
-                            class: 'button_no-margin',
+      <%= styled_button_tag class: 'button_no-margin',
                             data: {
+                              'project-storage-form-target': "selectProjectFolderButton",
                               'action': "project-storage-form#selectProjectFolder"
                             },
                             type: 'button' do %>
         <%= spot_icon('folder-open') %>
         <%= content_tag :span, t(:"storages.buttons.select_folder") %>
       <% end %>
+      <%= angular_component_tag 'op-storage-login-button',
+                                inputs: {
+                                  input: storage_login_input(@project_storage.storage)
+                                },
+                                style: 'display: inline-block',
+                                data: {
+                                  'project-storage-form-target': "loginButton"
+                                }
+      %>
       <span
         data-project-storage-form-target="selectedFolderText"
-      ></span>
+      >...</span>
     </div>
-    <% if @errors.present? && @errors.any? %>
+    <% if @errors&.any? { |error| error.attribute == :project_folder_id } %>
```

Comment:
I'm not sure why we need to filter errors by `project_folder_id`. I would presume we want to show errors generically for the project_storage?

Perhaps a note?

Reply:
there is currently only one possible error, but also if there were multiple, this very `span` should be only shown, if this particular error for the attribute `project_folder_id` is among the lists of errors.

Reply:
Got you, I see we have the `projectFolderIdValidationTarget` for the error setting from the stimulus controller. 👍🏾